<?xml version="1.0"?>
<launch>
    <!-- set camera to -->
    <arg name="camera" default="image_raw" />
    <!-- set camera id to -->
    <arg name="camera_id" default="left" />
    <!-- set camera name to -->
    <arg name="camera_name" default="cam" />
    <!-- frame_id for the camera -->
    <arg name="frame_id" default="$(arg camera_name)" />
    <!-- set camera fps to -->
    <arg name="set_camera_fps" default="30" />
    <arg name="w_mode" default="manual" />
    <arg name="w_red" default="1.58" />
    <arg name="w_green" default="1.0" />
    <arg name="w_blue" default="1.9" />
    <arg name="gain_mode" default="continuous" />
    <arg name="gain" default="24.0" />

    <!-- By default, calibrations are stored to file://${ROS_HOME}/camera_info/${NAME}.yaml
    To use your own fill this arg with the corresponding url, e.g.:
    "file:///$(find your_camera_package)/config/your_camera.yaml" -->
    <!-- <arg name="camera_info_url" default="file:///$(find offboard)/config/calibration.yaml" /> -->
    <arg name="camera_info_url" default="" />
    <!-- visualize on an image_view window the stream generated -->
    <arg name="visualize" value="true" />

    <!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
    <group ns="$(arg camera_name)">
        <node pkg="daheng_ros" type="daheng" name="$(arg camera_name)_stream" output="screen">
            <remap from="camera" to="$(arg camera)" />
            <param name="camera_id" type="string" value="$(arg camera_id)" />
            <param name="camera_name" type="string" value="$(arg camera_name)" />
            <param name="set_camera_fps" type="double" value="$(arg set_camera_fps)" />
            <param name="frame_id" type="string" value="$(arg frame_id)" />
            <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
            <param name="w_mode" type="string" value="$(arg w_mode)" />
            <param name="w_red" type="double" value="$(arg w_red)" />
            <param name="w_green" type="double" value="$(arg w_green)" />
            <param name="w_blue" type="double" value="$(arg w_blue)" />
            <param name="gain_mode" type="string" value="$(arg gain_mode)" />
            <param name="gain" type="double" value="$(arg gain)" />
        </node>

        <node if="$(arg visualize)" name="$(arg camera_name)_image_view" pkg="image_view" type="image_view">
            <remap from="image" to="$(arg camera)" />
        </node>
    </group>

</launch>
